File: plugins/kraken_sdk/ios/Classes/KrakenSDKPlugin.m
#import "Kraken.h"
#import "KrakenSDKPlugin.h"

@implementation KrakenSDKPlugin

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"kraken_sdk"
            binaryMessenger:[registrar messenger]];
  
  KrakenSDKPlugin* instance = [[KrakenSDKPlugin alloc] init];
  
  [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"setIsolateId" isEqualToString:call.method]) {
    NSDictionary* argsMap = call.arguments;
    NSString* isolateId = argsMap[@"isolateId"];
    if (isolateId != nil) {
      self.isolateId = isolateId;
    }
    result(nil);
  } else if ([@"getUrl" isEqualToString:call.method]) {
    Kraken* krakenInstance = [Kraken get:self.isolateId];
    if (krakenInstance != nil) {
      result([krakenInstance getUrl]);
    } else {
      result(nil);
    }
  } else {
    result(FlutterMethodNotImplemented);
  }
}

@end

File: plugins/kraken_sdk/macos/Classes/KrakenSDKPlugin.swift
import Cocoa
import FlutterMacOS

public class KrakenSDKPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "kraken_sdk", binaryMessenger: registrar.messenger)
    let instance = KrakenSDKPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
